contracts/core/src/contract.rs

```diff
@@ -121,6 +121,8 @@ pub fn query(deps: Deps<NeutronQuery>, _env: Env, msg: QueryMsg) -> ContractResu
             response: LAST_STAKER_RESPONSE.may_load(deps.storage)?,
         })?,
         QueryMsg::PauseInfo {} => query_pause_info(deps)?,
+        QueryMsg::TotalLSMShares {} => to_json_binary(&TOTAL_LSM_SHARES.load(deps.storage)?)?,
+        QueryMsg::FailedBatch {} => to_json_binary(&FAILED_BATCH_ID.may_load(deps.storage)?)?,
```

Comment:
Please define a `struct FailedBatchResponse { pub: failed_batch: Option<u128> }`, serialize data into it and then return